src/main/java/org/jboss/remoting3/remote/ClientConnectionOpenListener.java

```diff
@@ -401,7 +401,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         factoryOperator = and(ClientConnectionOpenListener.this.saslClientFactoryOperator, factoryOperator);
                         final SaslClient saslClient;
                         try {
-                            saslClient = configurationClient.createSaslClient(uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
+                            saslClient = createSaslClient(configurationClient, uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
```

Comment:
Looking into it `PriviliegedSaslClientFactory` fails the same way because its `createSaslClient()` is not run in a privileged way, https://github.com/iweiss/wildfly-elytron/blob/abd764fa33f735ba1850a7402cb1488d385bc0dc/src/main/java/org/wildfly/security/sasl/util/PrivilegedSaslClientFactory.java#L55.

Should this then be fixed on Elytron instead? What do you think?